8248: [Accessibility] Low Contrast in High Contrast mode on Stacktrace View

Suchitainf opened the pull request:
This PR is to fix the accessibility issue for High Contrast mode. Stracktrace View was not readable at all in dark mode.

Before the fix:
<img width="959" alt="image" src="https://github.com/user-attachments/assets/f12f1b7d-aa8a-4cce-8e0c-7731d5f9cc96">

After the fix:
<img width="959" alt="image" src="https://github.com/user-attachments/assets/8f7be98a-206b-4758-a1c2-8971d7ab1c55">

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Issue
 * [JMC-8248](https://bugs.openjdk.org/browse/JMC-8248): [Accessibility] Low Contrast in High Contrast mode on Stacktrace View (**Bug** - P4)


### Reviewers
 * [Alex Macdonald](https://openjdk.org/census#aptmac) (@aptmac - **Reviewer**)
 * [Brice Dutheil](https://openjdk.org/census#bdutheil) (@bric3 - Committer)


### Contributors
 * Marcus Hirt `<hirt@openjdk.org>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jmc.git pull/580/head:pull/580` \
`$ git checkout pull/580`

Update a local copy of the PR: \
`$ git checkout pull/580` \
`$ git pull https://git.openjdk.org/jmc.git pull/580/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 580`

View PR using the GUI difftool: \
`$ git pr show -t 580`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jmc/pull/580.diff">https://git.openjdk.org/jmc/pull/580.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jmc/pull/580#issuecomment-2309873797)

bric3 commented on application/org.openjdk.jmc.ui.common/src/main/java/org/openjdk/jmc/ui/common/util/ThemeUtils.java:
```diff
@@ -0,0 +1,30 @@
+import org.eclipse.swt.graphics.RGB;
+import org.eclipse.ui.PlatformUI;
+
+public class ThemeUtils {
```
**suggestion:** I believe the themes can change at runtime. `isDarkMode` will compute this value always. But to avoid the computation on each call, you could store a variable here and update this value each time the theme changes.

Something like:

```java
public class ThemeUtils {
  private static boolean isCurrentThemeDark = isDarkMode();

  static {
    PlatformUI.getWorkbench().getThemeManager().addPropertyChangeListener(() -> isCurrentThemeDark = isDarkMode())
  }

  public static boolean isDarkThemeTheme() { return isCurrentThemeDark; }
}
```

This way the value is updated anytime the theme changes. I'm not quite sure when to register the listener, I put it in static block as an example, but maybe this is not recommended in the Eclipse RCP programing idioms.

Note I kept the original name of `isDarkMode`, but feel free to rename to anything meaningful in your implementation.

Suchitainf commented:
/contributor add @thegreystone

aptmac reviewed (changes requested):
Brice has a good point here, applying themes can be dynamic so IS_DARK_MODE needs to be able to be updated.

Suchitainf commented:
Hi @bric3 & @aptmac , I have implemented the suggested change. I have also added the property listener in the static block. There is no issue with the eclipse development environment. I have tried to build JMC outside also and the build is working perfectly fine. But seems like spotbugs is not happy with updating static variable isCurrentThemeDark from the listener. Hence, there are test failures due to that. Please have a look at the changes and let me know if I am missing something.

Hope I understood the suggestion clearly and implemented accordingly.

bric3 commented on application/org.openjdk.jmc.ui.common/src/main/java/org/openjdk/jmc/ui/common/util/ThemeUtils.java:
```diff
@@ -1,22 +1,46 @@
 
 public class ThemeUtils {
+
+	private static boolean isCurrentThemeDark = isDarkMode();
```
**suggestion:** 

> But seems like spotbugs is not happy with updating static variable isCurrentThemeDark from the listener

I don't know if it's alright by JMC spotbug rules to write this global state like this

```suggestion
	private static AtomicReference<Boolean> isCurrentThemeDark = new AtomicReference<>(isDarkMode());
```

and then access it with `AtomicReference::get` / `AtomicReference::set`.

Suchitainf replied:
Thanks for the suggestion. I have tried fixing it with another simple approach and updated the PR with that change. Spotbugs is also happy with that. Can you please have a look? If it doesn't look good, I will try this AtomicReference approach. Thanks in advance.

bric3 replied:
@Suchitainf Your alternative approach is perfectly fine :)

bric3 commented on application/org.openjdk.jmc.ui.common/src/main/java/org/openjdk/jmc/ui/common/util/ThemeUtils.java:
```diff
@@ -33,15 +34,17 @@ public static boolean isDarkTheme() {
 	private static final double BRIGHTNESS_THRESHOLD = 0.5;
 
-	private static boolean isDarkMode() {
+	private static void isDarkMode() {
```
**nitpick:** I'd rename `isDarkMode` that better describe that it updates the state, e.g. 

```suggestion
	private static void computeCurrentThemeDarkModeStatus() {
```

Suchitainf replied:
Done.

Suchitainf commented:
Thanks @bric3  for approving the changes. 

@aptmac Can you please take a look now? I need one reviewer's approval before integrating the changes. Thanks in advance.

Suchitainf commented:
/integrate
